events/nats_connection.go

```diff
@@ -0,0 +1,152 @@
+package events
+
+import (
+	"bytes"
+	"context"
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
+	"strings"
+
+	"github.com/nats-io/nats.go"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/trace"
+	"go.uber.org/zap"
+)
+
+const (
+	base10 = 10
+
+	natsTracerName = tracerName + ":nats"
+)
+
+var _ Connection = (*NatsConnection)(nil)
+
+// NatsConnection implements Connection.
+type NatsConnection struct {
+	logger    *zap.SugaredLogger
+	tracer    trace.Tracer
+	conn      *nats.Conn
+	jetstream nats.JetStreamContext
+	cfg       NatsConfig
+}
+
+// Close gracefully drains the connection.
+func (c *NatsConnection) Close(ctx context.Context) error {
```

Comment:
Convention in Go is that `Close` generally takes no values: https://pkg.go.dev/io#Closer